From 2eb6812b4291a5d1ec38fb4b4b6ad3a59c4a8414 Mon Sep 17 00:00:00 2001
From: nsamsono <nsamsono@marvell.com>
Date: Sun, 29 Oct 2017 17:01:52 +0200
Subject: [PATCH 4/5] net/mrvl: add port speed capabilities to
 mrvl_dev_infos_get()

This is a workaround since we don't have yet MUSDK API to
retrieve it from HW

Change-Id: I760bf9d7e2f91181f1fb85490869dd0e893d9003
Signed-off-by: nsamsono <nsamsono@marvell.com>
---
 drivers/net/mrvl/mrvl_ethdev.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/mrvl/mrvl_ethdev.c b/drivers/net/mrvl/mrvl_ethdev.c
index d1633ba..496859a 100644
--- a/drivers/net/mrvl/mrvl_ethdev.c
+++ b/drivers/net/mrvl/mrvl_ethdev.c
@@ -867,6 +867,11 @@ struct mrvl_txq {
 mrvl_dev_infos_get(struct rte_eth_dev *dev __rte_unused,
 		    struct rte_eth_dev_info *info)
 {
+	info->speed_capa = ETH_LINK_SPEED_10M |
+			   ETH_LINK_SPEED_100M |
+			   ETH_LINK_SPEED_1G |
+			   ETH_LINK_SPEED_10G;
+
 	info->max_rx_queues = MRVL_PP2_RXQ_MAX;
 	info->max_tx_queues = MRVL_PP2_TXQ_MAX;
 	info->max_mac_addrs = MRVL_MAC_ADDRS_MAX;
-- 
1.9.1

